WordPress/src/main/java/org/wordpress/android/ui/stories/StoryComposerActivity.kt

```diff
@@ -510,39 +512,46 @@ class StoryComposerActivity : ComposeLoopFrameActivity(),
-                    val storyMediaFileData =
-                            saveStoryGutenbergBlockUseCase.buildMediaFileDataWithTemporaryIdNoMediaFile(
-                                    temporaryId = "$storyIndex-$frameIndex",
-                                    url = if (frame.source is FileBackgroundSource) {
-                                        (frame.source as FileBackgroundSource).file.toString()
-                                    } else {
-                                        (frame.source as UriBackgroundSource).contentUri.toString()
-                                    },
-                                    isVideo = (frame.frameItemType is VIDEO)
-                            )
+                    val storyMediaFileData = buildStoryMediaFileDataForTemporarySlide(
+                            frame,
+                            assignedTempId
+                    )
                     frame.id = storyMediaFileData.id
                     storyMediaFileDataList.add(storyMediaFileData)
                 }
-                // if the frame.id is populated, this should be an actual MediaModel mediaId so,
+                // if the frame.id is populated and is not a temporary id, this should be an actual MediaModel mediaId so,
                 // let's use that to obtain the mediaFile and then replace it with the temporary frame.id
                 else -> {
                     frame.id?.let {
-                        val mediaModel = mediaStore.getSiteMediaWithId(site, it.toLong())
-                        val mediaFile = fluxCUtilsWrapper.mediaFileFromMediaModel(mediaModel)
-                        mediaFile?.let {
-                            val storyMediaFileData =
-                                    saveStoryGutenbergBlockUseCase.buildMediaFileDataWithTemporaryId(
-                                            mediaFile = it,
-                                            temporaryId = "$storyIndex-$frameIndex"
-                                    )
-                            frame.id = storyMediaFileData.id
+                        if (it.startsWith(TEMPORARY_ID_PREFIX)) {
+                            val storyMediaFileData = buildStoryMediaFileDataForTemporarySlide(
+                                    frame,
+                                    it
+                            )
+                            // frame.id = storyMediaFileData.id
                             storyMediaFileDataList.add(storyMediaFileData)
+                        } else {
+                            val mediaModel = mediaStore.getSiteMediaWithId(site, it.toLong())
+                            val mediaFile = fluxCUtilsWrapper.mediaFileFromMediaModel(mediaModel)
+                            mediaFile?.let {
+                                val storyMediaFileData =
+                                        saveStoryGutenbergBlockUseCase.buildMediaFileDataWithTemporaryId(
+                                                mediaFile = it,
```

Comment:
could we give the parameter `it` a name so that the outer one isn't being shadowed? 🙏

Reply:
👍 done in 4d1eaa16d6